src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -81,36 +83,53 @@ public SourceInfo<string> GetConfigMonikerRange(FilePath file)
             return (null, monikers);
         }
 
-        private (Error? error, string[] monikers) GetFileLevelMonikersCore(FilePath file)
+        private (List<Error> errors, string[] monikers) GetFileLevelMonikersCore(FilePath file)
         {
+            var errors = new List<Error>();
             var (_, metadata) = _metadataProvider.GetMetadata(file);
 
             var configMonikerRange = GetConfigMonikerRange(file);
-            var configMonikers = _rangeParser.Parse(configMonikerRange);
+            var configedMonikers = _rangeParser.Parse(configMonikerRange);
 
-            if (!string.IsNullOrEmpty(metadata.MonikerRange))
+            if (!string.IsNullOrEmpty(metadata.MonikerRange) || metadata.Monikers.Length > 0)
             {
                 // Moniker range not defined in docfx.yml/docfx.json,
                 // user should not define it in file metadata
-                if (configMonikers.Length == 0)
+                if (configedMonikers.Length == 0)
                 {
-                    return (Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange), configMonikers);
+                    errors.Add(Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange));
+                    return (errors, configedMonikers);
+                }
+
+                // monikerRagen takes precedence over monikers since it is more likely from user configuration
+                if (!string.IsNullOrEmpty(metadata.MonikerRange) && metadata.Monikers.Length > 0)
+                {
+                    var source = metadata.Monikers.First().Source;
+                    if (source != null)
+                    {
+                        errors.Add(Errors.Versioning.DuplciateMonikerConfig(source));
+                    }
                 }
 
                 var fileMonikers = _rangeParser.Parse(metadata.MonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers).ToArray();
+                if (fileMonikers.Length == 0 && metadata.Monikers.Length > 0)
```

Comment:
>if (fileMonikers.Length == 0 && metadata.Monikers.Length > 0) [](start = 16, length = 61)

Hah, think about this case:
```
---
monikerRange: >1.0 < 1.0
monikers: 1.0
---
the final monikers will be `1.0`, and a `duplciate-moiniker` warning is reported with msg says `we take the monikerRange`?

Reply:
Nope, the final monikers will be empty, which is the output of `monikerRange`.
Take a look at the [check](https://github.com/dotnet/docfx/pull/5686/files#diff-afafc49d45fa0ff0eb00d49f86fc7084L91) in this PR

Reply:
The check here should be `if (!string.IsNullOrEmpty(metadata.MonikerRange) && metadata.Monikers.Length > 0)`.

Think about your case here, if `monikers` takes precedence, and `monikers` only contains invalid values. It would be the same

Reply:
if `monikers` takes precedence, and monikers only contains invalid values, then we take the invalid values and report `undefined` warnings

Reply:
`moniker-range-invalid` is an error.
And check the test added in my last commit, error `moniker-range-out-of-scope` is thrown, which is similar